### ISSUE TITLE
$parent.$current behaviour changes with indexed queries

### ISSUE THREAD

**Issue by userg** [orientechnologies/orientdb#8412]
## Expected behavior  
```
select id, $max, $maxRecord from ABC
let
$max = (select max($parent.$current.out('A').value) as maxVal),
$maxRecord = (select $parent.$current.out('A')[value = $parent.$current.$max.maxVal])
where id in [...]
```
This works as long as `id` in `ABC` is not indexed. 

## Actual behavior  
After adding a unique/non-unique index on `id`, $parent.$current result changes to `{key: , rid: }` format.  And the above query returns null for $max and $maxRecord.

**Comment by luigidellaquila** [comment]
Hi @userg 

Could you please provide a dataset (or a script) to reproduce the problem?

Thanks

Luigi

**Comment by userg** [comment]
Please use this script to reproduce
`
create class ABC extends V;
create property ABC.tid Integer;
create class XYZ extends V;
create property XYZ.value long;
create class A extends E;

let $abc = create vertex ABC set tid = 1;
let $xyz1 = create vertex XYZ set value = 100;
let $xyz2 = create vertex XYZ set value = 200;
create edge A from $abc to $xyz1;
create edge A from $abc to $xyz2;

-- this works (Returns result)
select tid, $maxvalues, $maxRecord from ABC
let
$maxvalues = (select max($parent.current.out('A').value) as mVal),
$maxRecord = (select $parent.current.out('A')[value in $parent.current.$maxvalues.mVal])
where tid in [1];

-- adding index
CREATE INDEX ABC.tid UNIQUE;

-- Same query (above) returns null
select tid, $maxvalues, $maxRecord from ABC
let
$maxvalues = (select max($parent.current.out('A').value) as mVal),
$maxRecord = (select $parent.current.out('A')[value in $parent.current.$maxvalues.mVal])
where tid in [1];
`

The same query on indexed attribute returns different response.

**Comment by misu200** [comment]
I think I ve the same issue. In Orient 3.0.5 my  following query  not returning the expected stuff compared to 2.2.30 on the same database ( and yes ..all the involved indexes are working good):

    SELECT $temp FROM Persoana let $temp = (select * from (select expand($parent.$current.out())) where denumire LIKE '%SERET%') WHERE [numeCanonic,dataNastere,adresa] LUCENE '(numeCanonic:(+("bulica") +("maren")))'  

Orient 2.2.30:
             "$temp": [
                "#32:92555"
            ]
Orient 3.0.5:
            "$temp": [
                
            ]

**Comment by luigidellaquila** [comment]
Hi @userg 

I just pushed a fix for this issue, it will be released with v 3.0.7

Thanks

Luigi
